src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -244,7 +235,10 @@ private String getGeneratedName(Type type) {
 
 			varName = Character.toLowerCase(varName.charAt(0)) + varName.substring(1);
 			if (plural) varName += "s";
-			return varName + "_" + nameCounts.compute(varName, (k, v) -> (v == null) ? 1 : v + 1);
+			String name = varName + "_" + nameCounts.compute(varName, (k, v) -> (v == null) ? 1 : v + 1);
+			renamedInvalidLocals.put(index, name);
```

Comment:
This isn't safe for non-parameter LVT indexes, they can and will be reused:
```java
public void method(int param) {
	if (...) {
		int value = ...
		...
	}

	if (...) {
		String otherValue = ...
		...
	}
}
```
would end up being remapped to
```java
public void method(int int_1) {
	if (...) {
		int int_2 = ...
		...
	}

	if (...) {
		String int_2 = ...
		...
	}
}
```
Perhaps they should only be added when `index < argCount`? You could also do `String[] renamedInvalidLocals  = new String[argCount]` then check if `index < renamedInvalidLocals.length`.

Reply:
>  You could also do `String[] renamedInvalidLocals = new String[argCount]` then check if `index < renamedInvalidLocals.length`.

I could do that, but I don't know if that's actually more efficient. This would mean, that I have to calculate the index in `visitLocalVariable`. Currently, the lvIndex needs to be calculated in `checkParameters`, but that gets calculated either way, since it's needed for the parameter name mapping method.

Reply:
Oh yeah double slot parameters would mean `argCount <= argLvSize`. It's probably easiest to just use `argLvSize` as the array size with holes (given there'll never be any locals which would use the holes). Saves having to do any more work for LVT indexes than is done already.